Sources/OpenTelemetrySdk/Trace/RecordEventsReadableSpan.swift

```diff
@@ -86,7 +87,15 @@ public class RecordEventsReadableSpan: ReadableSpan {
     /// The end time of the span.
     public private(set) var endTime: Date?
     /// True if the span is ended.
-    public private(set) var hasEnded: Bool = false
+    private var endAtomic = ManagedAtomic<Bool>(false)
```

Comment:
Why is it needed? it doesn't synchronize threads and the only thing this variable protects is whether the new attribute will be added after the span has ended. Not a big deal I think and may not worth an atomic variable on a critical path

Reply:
Is is also checked when adding events, I am finding this issue in my workflow because I am adding events from a secondary thread while the span ends

Reply:
you are not achieving any synchronization here, correct? Without atomic the only think will get better is when even was being added exactly at the moment span is being ended. Not much improvement over non-atomic?

Reply:
I modified the code a bit, so it doesn't have any atomic check in the critical path. 
It still uses the hasEnded atomic value to avoid the span to be ended twice from different threads and because is part of the public API and don't know how it is going to be used from outside.
But for the addAttribute() or addEvent() the code is now using the isRecording variable instead, that is only updated after end() has been called.

Reply:
great idea to split the check into two variables.